### PR TITLE
fix(hmacv4): improve obtaining info from Authorization header and setting date formats to match Amazon protocol better

### PR DESCRIPTION
**Pull request by RonaldHShi**
Change populating the host in the canonical headers in as part of "Task 1" in signing the HTTP request so there isn't a duplicate. Also, fix how the credential scope is created in case the date is already in the correct format. 

@philloooo

### CODE REVIEW

File: cdispyutils/hmac4/hmac4_auth_utils.py [hunk 1]
```diff
@@ -177,6 +177,11 @@ def get_request_date(req):
     for header in [constants.REQUEST_DATE_HEADER, 'date']:
         if header not in req.headers:
             continue
+        try: 
+            date = datetime.datetime.strptime(req.headers[header], '%Y%m%d')
+            break
+        except ValueError:
+            continue
         try:
```
Comment (philloooo) [inline cdispyutils/hmac4/hmac4_auth_utils.py]: can you add some documentation about your changes to the date header? Not very clear to me when looking at the code

Comment (RonaldHShi) [inline cdispyutils/hmac4/hmac4_auth_utils.py]: Oops yeah, I thought I needed that but it's totally unnecessary actually

File: cdispyutils/hmac4/hmac4_auth_utils.py [hunk 2]
```diff
@@ -188,7 +188,7 @@ def get_request_date(req):
         else:
             break
 
-    return date
+    return date.strftime("%Y%m%d")
```
Comment (philloooo) [inline cdispyutils/hmac4/hmac4_auth_utils.py]: revert this change too?

**Comment by RonaldHShi** [comment]
Improve error messages
Match Authorization header with regex's rather than splitting the string
Fix date format in certain parts of signing process to match AWS signing protocol better
r @philloooo

File: cdispyutils/hmac4/hmac4_auth_utils.py [hunk 3]
```diff
@@ -90,16 +91,19 @@ def generate_signature(secret_key, sig_string):
...
-        vals = re.split('\s|/|=', authorization_header)
-        if len(vals) < 10:
-            raise UnauthorizedError("Incorrect authentication format!")
-        return vals[2], vals[10]
+        signature = re.match(r'.*Signature=(\S*)', authorization_header).group(1)
+        access_key = re.match(r'.*Credential=(\S*?)\/.*', authorization_header).group(1)
+        return access_key, signature
     except:
+        raise AttributeError("No authentication provided or authorization header incorrect! Missing signature or credential")
         raise UnauthorizedError("No authentication provided!")
```
Comment (philloooo) [inline cdispyutils/hmac4/hmac4_auth_utils.py]: this need to be removed?
